File: 905-sort-array-by-parity/905-sort-array-by-parity.rs
use std::cmp::Ordering;

pub fn is_even(n: &i32)->bool{
    n%2 == 0
}

impl Solution {
    pub fn sort_array_by_parity(nums: Vec<i32>) -> Vec<i32> {
        let mut nums = nums;
        
        nums.sort_by(|a, b|{ 
            if (is_even(a) == is_even(b)){
                a.cmp(b)
            }else if (is_even(a)){
               Ordering::Less 
            }else {
                Ordering::Greater
            }
        });
        
        nums
    }
}

File: 905-sort-array-by-parity/905-sort-array-by-parity.py
class Solution:
    def sortArrayByParity(self, nums: List[int]) -> List[int]:
        pivot = 0
        
        for (i, n) in enumerate(nums):
            if n % 2 == 0:
                swap(nums, pivot, i)
                pivot+=1
        
        return nums
            
def swap(nums: List[int], i:int, j: int):
    tmp = nums[i]
    nums[i] = nums[j]
    nums[j] = tmp